File: app/src/main/java/com/example/energo_monitoring/model/DevicePressureTransducer.java
package com.example.energo_monitoring.model;

import com.example.energo_monitoring.model.api.DeviceInfo;

public class DevicePressureTransducer extends DeviceInfo {
    private String installationPlace;
    private String manufacturer;
    private String values;
    private String sensorRange;

    public DevicePressureTransducer(String name, int typeId, String installationPlace, String manufacturer, String values) {
        super(name, typeId);
        this.installationPlace = installationPlace;
        this.manufacturer = manufacturer;
        this.values = values;
    }

    public String getInstallationPlace() {
        return installationPlace;
    }

    public void setInstallationPlace(String installationPlace) {
        this.installationPlace = installationPlace;
    }

    public String getManufacturer() {
        return manufacturer;
    }

    public void setManufacturer(String manufacturer) {
        this.manufacturer = manufacturer;
    }

    public String getValues() {
        return values;
    }

    public void setValues(String values) {
        this.values = values;
    }

    public String getSensorRange() {
        return sensorRange;
    }

    public void setSensorRange(String sensorRange) {
        this.sensorRange = sensorRange;
    }

    @Override
    public int getFilledState() {
        if(!(installationPlace == null || installationPlace.isEmpty())
                && !(manufacturer == null || manufacturer.isEmpty())
                && !(sensorRange == null || sensorRange.isEmpty())
                && !(values == null || values.isEmpty()))
            return 2;
        else if(!(installationPlace == null || installationPlace.isEmpty())
                || !(manufacturer == null || manufacturer.isEmpty())
                || !(sensorRange == null || sensorRange.isEmpty())
                || !(values == null || values.isEmpty()))
            return 1;
        else
            return 0;
    }
}

File: app/src/main/java/com/example/energy_monitoring/checks/data/devices/DevicePressureTransducer.kt
package com.example.energy_monitoring.checks.data.devices

import androidx.room.Entity
import com.example.energy_monitoring.checks.data.api.DeviceInfo
import androidx.room.PrimaryKey

@Entity
class DevicePressureTransducer : DeviceInfo() {
    
    @PrimaryKey(autoGenerate = true)
    override var id = 0

    var installationPlace = Field("")
    var manufacturer = Field("")
    var values = Field("")
    var sensorRange = Field("")
    var comment = Field("")

//    val filledState: Int
//        get() = if (!(installationPlace == null || installationPlace!!.isEmpty())
//            && !(manufacturer == null || manufacturer!!.isEmpty())
//            && !(sensorRange == null || sensorRange!!.isEmpty())
//            && !(values == null || values!!.isEmpty())
//        ) 2 else if (!(installationPlace == null || installationPlace!!.isEmpty())
//            || !(manufacturer == null || manufacturer!!.isEmpty())
//            || !(sensorRange == null || sensorRange!!.isEmpty())
//            || !(values == null || values!!.isEmpty())
//        ) 1 else 0
}